p2p/host.go

```diff
@@ -295,6 +299,11 @@ func New(
 			cfg.AutoNATServer.PeerMax,
 			cfg.AutoNATServer.ResetPeriod),
 		libp2p.ConnectionGater(g),
+		libp2p.WithFxOption(fx.Invoke(func(ids identify.IDService) {
+			identifyConn = func(c network.Conn) {
+				ids.IdentifyConn(c)
+			}
+		})),
```

Comment:
This feels like a hack to me, especially since later down we panic if this doesn't work.

Reply:
This should not fail unless `fx` DI library that `libp2p` uses is broken.
I can add a `panic` call to clarify that in case if this doesn't work.
Will try to get rid of this function variable

Reply:
It now uses a variable of `identity.IDService` type and returns an error instead of panicking if for whatever reason `fx` fails to invoke the function